src/js/components/conversation.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import { connect } from 'react-redux';
+
+import { createMarkup } from 'utils/html';
+
+import { MessageComponent } from 'components/message.jsx';
+
+export class ConversationComponent extends Component {
+    _scrollToBottom() {
+        let container = findDOMNode(this);
+        let logo = this.refs.logo;
+        let scrollTop = container.scrollHeight - container.clientHeight - logo.clientHeight;
+        container.scrollTop = scrollTop;
+    }
+
+    componentDidMount() {
+        setTimeout(this._scrollToBottom.bind(this));
+    }
+
+    componentDidUpdate() {
+        setTimeout(this._scrollToBottom.bind(this));
+    }
+
+    render() {
+        const messages = this.props.conversation.messages.map((message) => <MessageComponent key={ message._id } {...message} />);
```

Comment:
How does string parsing work in a jsx file? i.e. how does it know that `<MessageComponent key={ message._id } {...message} />`, and the stuff in between `return ()` are strings?